workflow/executor/executor.go

```diff
@@ -356,27 +356,28 @@ func (we *WorkflowExecutor) SaveParameters() error {
 }
 
 // SaveLogs saves logs
-func (we *WorkflowExecutor) SaveLogs() error {
+func (we *WorkflowExecutor) SaveLogs() (*wfv1.Artifact, error) {
 	if we.Template.ArchiveLocation == nil || we.Template.ArchiveLocation.ArchiveLogs == nil || !*we.Template.ArchiveLocation.ArchiveLogs {
-		return nil
+		return nil, nil
 	}
 	log.Infof("Saving logs")
 	mainCtrID, err := we.GetMainContainerID()
 	if err != nil {
-		return err
+		return nil, err
 	}
 	tempLogsDir := "/argo/outputs/logs"
 	err = os.MkdirAll(tempLogsDir, os.ModePerm)
 	if err != nil {
-		return errors.InternalWrapError(err)
+		return nil, errors.InternalWrapError(err)
 	}
 	fileName := "main.log"
 	mainLog := path.Join(tempLogsDir, fileName)
 	err = we.RuntimeExecutor.Logs(mainCtrID, mainLog)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	art := wfv1.Artifact{
+		Name:             fmt.Sprintf("%s/logs", we.PodName),
```

Comment:
Wont the slash in the name cause issues when referencing the artifact?

Reply:
We actually restrict characters in what user can use in a name to DNS chars. So we would be violating our own rules. I think hardwiring `main-log` might be better. podname is not easily predicted by a user

Reply:
I wanted to avoid hardcoding name, since the user might define output with the same name and it would cause a conflict. 

Used pod name instead since it is same as node id, so node logs will be available in `<node-id>/logs`. 

If you think it is ok to use `node-id` convention I would propose `<node-id>-main-logs`.  Don't mind using just `main-logs` let me know which option is better.

Reply:
I do recognize that it could potentially collide with user defined names, but I think in this case hardwiring name is fine because:
1. we already do not allow collisions with the name `main` and `wait` for container names.
2. we can catch it during linting so that it's not a runtime error.
3. it's valuable to have predictability in the name (in case user wants to actually reference the log artifact)
4. using <containername>-logs is probably not something user will decide for their name anyways
5. the easy workaround is for user to choose a different name

Reply:
Sounds good. Changed name to `main-logs` .